pkgs/shells/powershell/default.nix

```diff
@@ -22,7 +23,7 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [ autoPatchelfHook makeWrapper ];
-  propagatedBuildInputs = [ libunwind libuuid icu curl cacert less openssl ];
+  propagatedBuildInputs = [ cacert less ] ++ libraries;
```

Comment:
Why are all those packages in propagatedBuildInputs?

Reply:
Perhaps I don't fully understand that part yet. The intention here was to submit that those are runtime dependencies, and not explosively build-time ones. If there's another, correct way to do this, please let me know.

Reply:
Nix figures out runtime dependencies by inspecting what files are referenced by a derivation.
Sometimes it cannot do this if for example an program do PATH search to run a different program.
In this case we use for example makeWrapper to extend the programs PATH.
`propagatedBuildInputs` will only be pulled if it this package is used as an build input for a different package. 
This is also the case for nix-shell by accident, which is why people sometimes think `propagatedBuildInputs` affect the runtime dependencies. Python is an exception to this rule since it uses `propagatedBuildInputs` in its own wrapper to extend the PYTHONPATH to contain all required python libraries used in an python script.

Reply:
less is already covered by the PAGER variable. Where is cacert used in powershell?

Reply:
As for `PAGER`, this is because there's a bug handling PAGER with arguments in the current release and there's a comment in this in the original nix file for PowerShell:

```nix
# TODO: remove PAGER after upgrading to v6.1.0-preview.1 or later as it has been addressed in
# https://github.com/PowerShell/PowerShell/pull/6144
```

Without this setting, current version will fail commands like `Help`:

```
PS /home/yrashk/Projects/nix/nixpkgs> help                                                                                                                                                                          
Get-Command : The term 'less -R' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

Once 6.1 is released, this can be removed.

Reply:
I've simplified build inputs to make this a much cleaner derivation as per your guidance.